pyanaconda/ui/tui/spokes/network.py

```diff
@@ -154,100 +158,96 @@ def refresh(self, args=None):
         EditTUISpoke.refresh(self, args)
 
         summary = self._summary_text()
-        self._window += [TextWidget(summary), ""]
+        self.window.add_with_separator(TextWidget(summary))
         hostname = _("Host Name: %s\n") % self.data.network.hostname
-        self._window += [TextWidget(hostname), ""]
+        self.window.add_with_separator(TextWidget(hostname))
         current_hostname = _("Current host name: %s\n") % network.current_hostname()
-        self._window += [TextWidget(current_hostname), ""]
+        self.window.add_with_separator(TextWidget(current_hostname))
 
         # if we have any errors, display them
         while len(self.errors) > 0:
-            self._window += [TextWidget(self.errors.pop()), ""]
-
-        def _prep(i, w):
-            """ Mangle our text to make it look pretty on screen. """
-            number = TextWidget("%2d)" % (i + 1))
-            return ColumnWidget([(4, [number]), (None, [w])], 1)
+            self.window.add_with_separator(TextWidget(self.errors.pop()))
 
         _opts = [_("Set host name")]
         for devname in self.supported_devices:
             _opts.append(_("Configure device %s") % devname)
-        text = [TextWidget(o) for o in _opts]
-
-        # make everything presentable on screen
-        choices = [_prep(i, w) for i, w in enumerate(text)]
-        displayed = ColumnWidget([(78, choices)], 1)
-        self._window += [displayed, ""]
+        text_widgets = [TextWidget(o) for o in _opts]
+
+        self._container = ListColumnContainer(1, columns_width=78, spacing=1)
+        self._container.add(text_widgets[0], callback=self._set_hostname_callback)
+
+        for i, w in enumerate(text_widgets[1:]):
+            self._container.add(w, callback=self._configure_network_interface, data=i)
+
```

Comment:
This method be simplified. Add the items to the container directly, we don't need the extra structures anymore.